File: 513_Find_Bottom_Left_Tree_Value.cpp
#include<bits/stdc++.h>

using namespace std;

struct TreeNode {
    int val;
    TreeNode* left;
    TreeNode* right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};


class Solution {
public:
    int findBottomLeftValue(TreeNode* root) {
        queue<TreeNode*> q; q.push(root);
        int res;
        while (!q.empty()){
            int s = q.size();
            for (int i = 0; i < s; ++i){
                auto t = q.front(); q.pop();
                if (i == 0) res = t->val;
                if (t->left) q.push(t->left);
                if (t->right) q.push(t->right);
            }
        }
        return res;
    }
};

int main(){
    Solution solution;
    TreeNode* root = new TreeNode(2);
    root->left = new TreeNode(1);
    root->right = new TreeNode(3);
    cout << solution.findBottomLeftValue(root) << endl;
    return 0;
}

File: 513_Find_Bottom_Left_Tree_Value.py
class Tree(object):
    def __init__(self, x):
    self.val = x
    self.left = None
    self.right = None


class Solution(object):
    def findBottomLeftValue(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        v, level = None, [root]
        while root and level:
            v = [x.val for x in level][0]
            temp = [(x.left, x.right) for x in level]
            level = [x for lr in temp for x in lr if x]
        return v
